packages/haul-core/src/server/setupCompilerRoutes.ts

```diff
@@ -78,8 +76,8 @@ export default function setupCompilerRoutes(
         );
 
         if (!hasRunAdbReverse && platform === 'android') {
-          await runAdbReverse(runtime, port);
           hasRunAdbReverse = true;
```

Comment:
it hasn't yet :D maybe worth changing the variable name if this is intended?

Reply:
ESLint: possible race condition if `hasRunAdbReverse` is changed after async operation.

Reply:
I'm fine as long as it makes sense in this scope (I didn't dive into how `hasRunAdbReverse` is used), just wanted to point it out

Reply:
`hasRunAdbReverse` is only to make sure `runAdbReverse` is run only once. If it fails we print error and let the user figure out what to do. We definitely don't want to run `runAdbReverse` multiple times. 

If `hasRunAdbReverse` is set after the `runAdbReverse` there is a chance that it will be called 2 times or more, since the next request will have a old `hasRunAdbReverse` value - not updated yet.